src/solvers/dgmulti/sbp.jl

```diff
@@ -313,23 +315,38 @@ function CartesianMesh(dg::DGMultiPeriodicFDSBP{NDIMS}) where {NDIMS}
   EToV = NaN # StartUpDG.jl uses size(EToV, 1) for the number of elements, this lets us reuse that.
   FToF = []
 
-  xyz = xyzq = rd.rst # TODO: DGMulti; extend to mapped domains
+  xyz = xyzq = rd.rst
   xyzf = ntuple(_ -> [], NDIMS)
   wJq = diag(rd.M)
 
   # arrays of connectivity indices between face nodes
   mapM = mapP = mapB = []
 
   # volume geofacs Gij = dx_i/dxhat_j
-  # TODO: DGMulti; extend to mapped domains
+  coord_diffs = coordinates_max .- coordinates_min
+
+  # Periodic SBP operators do not include one endpoint. We account for this by adding
+  # `h`` when estimating `factor`, which is the size of the "reference interval".
+  D = rd.approximationType
+  xmin = SummationByPartsOperators.xmin(D)
+  xmax = SummationByPartsOperators.xmax(D)
+  factor = (xmax - xmin)
```

Comment:
You assume that the size of the reference interval is `2` (with reference interval `[-1, 1]`), correct? If so, I would propose to include the scaling already when constructing the `dg::DGMulti` instance, also for periodic operators. Thus,
https://github.com/trixi-framework/Trixi.jl/blob/2e221fc49bf4deba4e072126358f277f775766c0/src/solvers/dgmulti/sbp.jl#L51-L70 
should basically become
```julia
 # StartUpDG assumes nodes from -1 to +1. Thus, we need to re-scale everything. 
 # We can adjust the grid spacing as follows. 
 xmin = SummationByPartsOperators.xmin(D) 
 xmax = SummationByPartsOperators.xmax(D) 
 factor = 2 / (xmax - xmin) 
 @. nodes_1d = factor * (nodes_1d - xmin) - 1 
 @. weights_1d = factor * weights_1d 
```
Then, we will probably need to update some code here. For example, we can't just set `xyz = xyzq = rd.rst` anymore but need to include some scaling from `[-1, 1]` to the given domain with `coordinates_min, coordinates_max`.

Reply:
I'm OK with rescaling all SBP operators to [-1, 1] prior to constructing `MeshData`, but there's no requirement on that. One could have an arbitrarily sized reference element so long as the geometric change of variables factors in `MeshData` are appropriately defined. 

That said, if you think it's better to rescale to [-1, 1] (for example, for consistency purposes) I'm also happy to do so.

Reply:
Ah, I stand corrected - `StartUpDG.jl` does assume nodes in [-1, 1], but this is only used when determining face nodes. Moreover, since `StartUpDG.RefElemData` is specialized for `AbstractDerivativeOperators`, these routines which assume nodes are in [-1, 1] are never called. 

I've gone ahead and rescaled nodes to [-1, 1] in this PR, but for future reference, `StartUpDG.jl` can accommodate nodes in arbitrary intervals in the future.